confmap/expand.go

```diff
@@ -111,22 +113,43 @@ func (mr *Resolver) findAndExpandURI(ctx context.Context, input string) (any, bo
 	if uri == input {
 		// If the value is a single URI, then the return value can be anything.
 		// This is the case `foo: ${file:some_extra_config.yml}`.
-		return mr.expandURI(ctx, input)
+		ret, ok, err := mr.expandURI(ctx, input)
+		if err != nil {
+			return input, false, err
+		}
+
+		raw, err := ret.AsRaw()
+		if err != nil {
+			return input, false, err
+		}
+
+		return raw, ok, nil
 	}
 	expanded, changed, err := mr.expandURI(ctx, uri)
 	if err != nil {
 		return input, false, err
 	}
-	repl, err := toString(expanded)
+
+	var repl string
+	if internal.StrictlyTypedInputGate.IsEnabled() {
+		repl, err = expanded.AsString()
```

Comment:
Can we add a test for this scenario?

Reply:
The end to end tests with `TargetFieldInlineString` test this. What kind of tests would you like to see?

Reply:
If the e2e test handles this we're good